src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,196 @@
+import { useEffect, useState } from 'react';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { Loader } from '../Loader';
+import { PersonLink } from '../PersonLink';
+import { SearchLink } from '../SearchLink';
+import { useSearchParams } from 'react-router-dom';
+
+type Props = {
+  setShowFilters: (arg: boolean) => void;
+};
+
+export const PeopleTable: React.FC<Props> = ({ setShowFilters }) => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [searchParams] = useSearchParams();
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('century');
+
+  const getSortParams = (value: string) => ({
+    sort: sort === value && order === 'desc' ? null : value,
+    order: sort === value && order === null ? 'desc' : null,
+  });
+
+  const sortFilters = ['Name', 'Sex', 'Born', 'Died'];
+
+  const handleError = (message: string) => {
+    setErrorMessage(message);
+    setTimeout(() => setErrorMessage(''), 3000);
+  };
+
+  useEffect(() => {
+    const getPeopleFromServer = async () => {
+      try {
+        setLoading(true);
+        const peopleFromServer = await getPeople();
+
+        const peopleWithParents = peopleFromServer.map(person => ({
+          ...person,
+          mother: peopleFromServer.find(
+            parent => parent.name === person.motherName,
+          ),
+          father: peopleFromServer.find(
+            parent => parent.name === person.fatherName,
+          ),
+        }));
+
+        setPeople(peopleWithParents);
+        setShowFilters(true);
+      } catch {
+        handleError('Something went wrong');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    getPeopleFromServer();
+  }, []);
+
+  function sortPeople(peopleToSort: Person[]): Person[] {
```

Comment:
pass sort param and order as arguments to the function and move it out of the component

Reply:
Fixed.